sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubProducer/EventHubProducerLiveTests.cs

```diff
@@ -181,5 +181,302 @@ await using (var producer = client.CreateProducer())
                 }
             }
         }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubProducer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProducerCanSendSingleZeroLengthEvent()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                await using (var producer = client.CreateProducer())
+                {
+                    var events = new[] { new EventData(new byte[0]) };
+                    Assert.That(async () => await producer.SendAsync(events), Throws.Nothing);
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubProducer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProducerCanSendSingleLargeEvent()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                await using (var producer = client.CreateProducer())
+                {
+                    // Actual limit is 1046520 for a single event
+                    var events = new[] { new EventData(new byte[1000000]) };
+                    Assert.That(async () => await producer.SendAsync(events), Throws.Nothing);
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubProducer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProducerCannotSendSingleEventLargerThanMaximumSize()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                await using (var producer = client.CreateProducer())
+                {
+                    // Actual limit is 1046520 for a single event
+                    var events = new[] { new EventData(new byte[1100000]) };
+                    Assert.ThrowsAsync<TrackOne.MessageSizeExceededException>(async () => await producer.SendAsync(events));
```

Comment:
Can we refactor this to use the constraint format, please?   I'd like to keep consistent within the tests.

```csharp
Assert.That(async () => async () => await producer.SendAsync(events), Throws.InstanceOf<TrackOne.MessageSizeExceededException>());
```

Reply:
Also... I'm taking a note to catch/wrap that exception into a new `MessageSizeExceededException` so that the `TrackOne` types are not leaked to the public scope.

Reply:
Refactored.

Reply:
`MessagingEntityNotFoundException` and `ReceiverDisconnectedException` should be taken into account as well.